web/docs/language/features.md

```diff
@@ -443,6 +443,149 @@ import { isPrismaError, prismaErrorToHttpError } from '@wasp/utils.js'
+  Executor specific options to use when enquing jobs.
+
+### JavaScript API
+
+#### Invocation
+##### `import`
+
+```js
+import { mySpecialJob } from '@wasp/jobs/mySpecialJob.js'
+```
+
+##### `submit(jobArgs, options)`
+- ###### `jobArgs: object` (optional)
+- ###### `options: object` (optional)
+
+Enqueues a `job` to be executed by an executor, optionally passing in job arguments and executor enqueue options.
+
+```js
+const submittedJob = await mySpecialJob.submit({ job: "args" })
+```
+
+##### `delay(startAfter)` (optional)
+- ###### `startAfter: int | string | Date` (required)
+
+Delaying the invocation of the job handler. The delay can be one of:
+- Integer: number of seconds to delay. [Default 0]
+- String: ISO date string to run at.
+- Date: Date to run at.
+
+```js
+const submittedJob = await mySpecialJob.delay(10).submit({ job: "args" })
+```
+
+#### Tracking
+The return value of `submit()` is an instance of `SubmittedJob`, that minimally contains:
+- `jobId`: A getter returning the UUID String ID for the job in that executor.
+- `jobName`: A getter returning the name of the job you used in your `.wasp` file. **NOTE**: See considerations below.
+- `executorName`: A getter returning the name of the job executor.
+
+There will also be namespaced, job executor-specific objects.
+- For pg-boss, you may access: `pgBoss`
+  - **NOTE**: no arguments are necessary, as we already applied the `jobId` in the available functions.
+  - `details()`: pg-boss specific job detail information.
+  - `cancel()`: attempts to cancel a job. [Ref](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#cancelid)
+  - `resume()`: attempts to resume a canceled job. [Ref](https://github.com/timgit/pg-boss/blob/master/docs/readme.md#resumeid)
+
+### Considerations
+
+There are job executor-specific considerations to keep in mind when making your selections. We will continue to offer additional job executor options that attempt to balance their respective pros and cons.
+
+#### pg-boss
+- Wasp starts pg-boss alongside your web server's application, where both are simultaneously operational. Accordingly, pg-boss and your web-server share the same NodeJS event loop, making it unsuitable for CPU-intensive tasks.
+  - Wasp does not support independent, horizontal scaling of pg-boss-only applications, nor starting them as separate workers/processes/threads.
+- The job name/identifier in your `.wasp` file is the same name that will be used in the `name` column of pg-boss tables. If you change a name that had a `schedule` associated with it, pg-boss will continue scheduling those jobs but they will have no handlers associated, and will thus become stale and expire. You can remove the applicable row from the `schedule` table in the `pgboss` schema of your database, or start a NodeJS REPL in your server context and run:
+  ```js
+  let { boss } = await import(`./src/jobs/core/pgBoss/pgBoss.js`)
+  boss.unschedule("mySpecialJob")
+  ```
```

Comment:
Great, should we merge this with PgBoss section that you also have above?

Reply:
Yeah, we can, I was on the fence on this one be it standalone vs a separate "watch out for", but combined makes sense.